GitCommands/PathUtil.cs

```diff
@@ -136,7 +136,7 @@ public static bool TryConvertWindowsPathToPosix([NotNull] string path, out strin
                 return false;
             }
 
-            posixPath = "/" + directoryInfo.FullName.ToPosixPath().Remove(1, 1);
+            posixPath = ("/" + directoryInfo.FullName.ToPosixPath().Remove(1, 1)).QuoteIfContainsWhiteSpace();
```

Comment:
I still feel that quoting shouldn't happen at this level, as we risk the chance of double-quoting. It's safest to quote at the time we compose the command line.

Reply:
Read the code again.  https://github.com/gitextensions/gitextensions/pull/5796/files#diff-534abaafd975edbb53b90514103d23a1R236 

Also find all references of this method.

Reply:
https://github.com/gitextensions/gitextensions/blob/1a258a2826a19d3139ff219a23318b7c1cfdc7a7/GitUI/CommandsDialogs/FormBrowse.cs#L2877

Only use of this method.

Reply:
I could also quote with ' instead of ".  Then I could roll it into ToPosix.

someCMD "blaaa '/folder/with spaces/and such/file.code'"

Reply:
I really wanted to make ToPosixPath do all the work.  

I think using ' when there are spaces in the path will work and we use " for quoting commands and args.  I can add a duplicate method called SingleQuoteIfContainsWhiteSpace that does the same just using '.  
```bash
cd '/e/box sync/' 
```
does work.  We won't be fighting " issues and only paths will have '.  

The fun part is if a filename actually contains " or '

Reply:
I know what you're trying to do, but it's different to what's done everywhere else. Why should `ToPosixPath` work differently to `TryConvertWindowsPathToPosix`?

The double quoting risk is that we currently compose command lines with code resembling `path.Quote()` or `path.ToPosixPath().Quote()`. We never have the string `path` be quoted while it's in flight. It's always done right when composing the command line.

It's easier when all paths in flight have the same convention.

Also, `System.IO` never gives you a quoted path string. There's no precedent for this.

If you want to pursue this, I suggest you do so in a future PR and do it across the board. We've talked before about having a `RepoPath` `struct` that wraps a `string` (possibly a root path too) and provides semantics (i.e. it's guaranteed to be a relative path within the repo). Perhaps there's something you can do with types that's less error prone. `string` is such a meaningless type for this kind of thing.

Reply:
Sure. I most definitely was going for an across-the-board change and was just trying to manage it all. 

Posix file paths support spaces by either using ```\ ``` or quotes(' or "). If you need to escape a character then you use \. So ```file made by John O'Mally``` would be a posix file name of ```'file made by John O\'Mally'```

Reply:
Totally agree all these rules should be captured and made correct and convenient.

Reply:
> System.io.path doesn't quote because it was built against windows and doesn't do posix paths.

No. It doesn't quote because the _path_ doesn't actually contain quotes.

The same is true in POSIX.

```
$ pwd
/c/Program Files (x86)
```

No quotes.

Quotes are required by the shell. If you use Bash, there's one set of quote rules. Other shells may have other rules.

```
$ file $(pwd)
/c/Program: cannot open `/c/Program' (No such file or directory)
Files:      cannot open `Files' (No such file or directory)
(x86):      cannot open `(x86)' (No such file or directory)

$ file "$(pwd)"
/c/Program Files (x86): directory
```

cmd.exe also uses quotes when you enter files. If you ask for a path, it too doesn't add quotes.

```
C:\Program Files (x86)>cd
C:\Program Files (x86)
```

We should not have quotes in strings that represent paths in memory. They should only be applied when we formulate command lines.

If you introduce a custom path type (as mentioned earlier) you can have an `ArgumentBuilder` extension add them automatically and consistently. For what you want I think maybe `PosixPath` would make sense. We could also have `RepoPath` as previously discussed.

Reply:
Got it.  Definitely was going to use ArgumentBuilder and provide those types.  I like the names.  I'll see what I can come up with.

Reply:
I'm not too worried about ```/c:/git/project 1/file```  It is just a nice to have.